File: DSAlgorithms/Leetcode/c++/arrays_hash/187-DNA-seq.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
       std::unordered_set<std::string> a_set;
       std::unordered_set<std::string> answer_set;
       std::vector<std::string> result;
       int s_size = s.size();
       for(int left = 0; left < (s_size - 9); left++)
       {
           std::string current = s.substr(left, 10);
           std::unordered_set<std::string>::iterator find_iterator = a_set.find(current);
           if(find_iterator != a_set.end())
                answer_set.insert(current);
            a_set.insert(current);
       }
       for(auto item: answer_set)
            result.push_back(item);
        return result;
    }
};

File: dsa_algo/leetcode/ts/src/arrays_hash/187-DNA-seq.ts
const findRepeatedDnaSequences = (s: string): string[] =>{
    const a_set = new Set<string>();
    const answer_set = new Set<string>();
    const result: string[] = [];
    const s_size: number = s.length;
    for(let left = 0; left < (s_size - 9); left++)
    {
        let current: string = s.substr(left, 10);
        if(a_set.has(current))
            answer_set.add(current);
        a_set.add(current);
    }
    answer_set.forEach(item => {
        result.push(item);
    })
    return result;
};